### PR TITLE
Implement access to neuron global variables.

### PR DESCRIPTION
**Pull request by 1uc**
What's meant is things like `celsius`.

### CODE REVIEW

File: src/codegen/codegen_neuron_cpp_visitor.cpp [hunk 1]
```diff
@@ -596,7 +596,7 @@ std::string CodegenNeuronCppVisitor::get_variable_name(const std::string& name,
     if (iter != info.neuron_global_variables.end()) {
         std::string ret;
         if (use_instance) {
-            ret = "*(inst->";
+            ret = "*(inst.";
```
Comment (ohm314) [inline src/codegen/codegen_neuron_cpp_visitor.cpp:599]: was that an error before?
by the way, why not use the `->` operator and drop the `*(...)` instead?

File: test/usecases/parameter/simulate.py [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
+celsius_hoc = h.Vector().record(s(0.5)._ref_range_celsius_NeuronVariables)
+
+h.stdinit()
+h.tstop = 5.0 * ms
+h.run()
+
+celsius = np.array(celsius_hoc.as_numpy())
+assert celsius[-1] == 6.3, f"{celsius[-1]=}"
+
+
```
Comment (1uc) [inline test/usecases/parameter/simulate.py:24]: We might need to configure the formatter to run `black` of these files.

File: test/usecases/parameter/neuron_variables.mod [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+PARAMETER {
+    celsius = 123.456
+}
+
+ASSIGNED {
+    range_celsius
+}
+
+BREAKPOINT {
+  range_celsius = celsius
```
Comment (1uc) [inline test/usecases/parameter/neuron_variables.mod:15]:
```suggestion
    range_celsius = celsius
```

File: src/codegen/codegen_neuron_cpp_visitor.cpp (hunk 1 above)
Comment (ohm314) [reply src/codegen/codegen_neuron_cpp_visitor.cpp:599]: ah.. duh.. yes obviously....

**Review by ohm314 — APPROVED** [review]

**Pull request merged**